runelite-client/src/main/java/net/runelite/client/plugins/wikilookup/WikiLookupPlugin.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2018, Tanner <https://github.com/Reasel>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.wikilookup;
+
+import com.google.common.eventbus.Subscribe;
+import net.runelite.api.Client;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.input.KeyListener;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.menus.MenuManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.LinkBrowser;
+import net.runelite.client.util.Text;
+import javax.inject.Inject;
+import java.awt.event.KeyEvent;
+
+@PluginDescriptor(
+	name = "Wiki Lookup",
+	enabledByDefault = false
+)
+public class WikiLookupPlugin extends Plugin implements KeyListener
+{
+	private static final String WIKI = "Wiki";
+
+	private static final int HOTKEY = KeyEvent.VK_SHIFT;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private MenuManager menuManager;
+
+	@Inject
+	private KeyManager keyManager;
+
+	private boolean hotKeyPressed = false;
+
+	@Override
+	public void keyTyped(KeyEvent e)
+	{
+
+	}
+
+	@Override
+	public void keyPressed(KeyEvent e)
```

Comment:
We now even have Hotkey class that you can use instead of this

Reply:
Well then I guess I have some work to do!

Reply:
I think we do not want to use that since I am trying to detect when the shift key is held and a menu is opened NOT just when the hotkey is pressed like in the Screenshot plugin.